docs/azure-openai-integration.md

```diff
@@ -0,0 +1,139 @@
+
+### Getting AI Suggestions
+1. After configuration, "AI Suggestion" buttons will appear on each risk card
+2. Click the "AI Suggestion" button for any risk you want to analyze
+3. Wait for the AI to generate the analysis (typically 2-10 seconds)
+4. Review the comprehensive suggestions provided
+5. Use the close button to hide suggestions when done
+
+## API Endpoints
+
+### POST /api/ai-suggestion
+Generate AI-powered risk analysis suggestions.
+
+**Request Body:**
+```json
+{
+  "risk_name": "detect_vusb_devices",
+  "risk_description": "vUSB devices connected to VMs",
+  "risk_data": [...],
+  "risk_level": "blocking",
+  "azure_endpoint": "https://your-resource.openai.azure.com/",
+  "api_key": "your-api-key",
+  "deployment_name": "gpt-4"
+}
+```
+
+**Response:**
+```json
+{
+  "success": true,
+  "suggestion": "AI-generated suggestion text...",
+  "risk_name": "detect_vusb_devices"
+}
+```
+
+### POST /api/test-azure-openai
+Test Azure OpenAI connection with provided credentials.
+
+**Request Body:**
+```json
+{
+  "azure_endpoint": "https://your-resource.openai.azure.com/",
+  "api_key": "your-api-key",
+  "deployment_name": "gpt-4"
+}
+```
+
+**Response:**
+```json
+{
+  "success": true,
+  "message": "Connection test successful",
+  "response": "Connection successful"
+}
+```
+
+## Security Considerations
+
+- **API Key Handling**: API keys are only stored in browser memory during the session
+- **No Persistent Storage**: Credentials are not saved to disk or databases
+- **HTTPS Required**: Azure OpenAI endpoints use HTTPS for secure communication
+- **Client-Side Configuration**: All configuration is done client-side for security
+
+## Technical Implementation
+
+### Backend Components
+- **AzureOpenAIService**: Handles Azure OpenAI API communication
+- **API Models**: Pydantic models for request/response validation
+- **API Endpoints**: FastAPI routes for AI functionality
+
+### Frontend Components
+- **Configuration UI**: Bootstrap-based configuration form
+- **JavaScript Functions**: Handle API calls and UI updates
+- **CSS Styling**: Consistent with existing application theme
+
+## Dependencies
+
+- **openai**: Python OpenAI library for Azure OpenAI integration
+- **pydantic**: For API request/response validation
+- **fastapi**: For API endpoint implementation
+
+## Error Handling
+
+The integration includes comprehensive error handling for:
+- Invalid credentials or endpoints
+- Network connectivity issues
+- API rate limiting
+- Service unavailability
+- Invalid deployment names
+
+Errors are displayed to users with clear, actionable messages.
+
+## Future Enhancements
+
+Potential future improvements could include:
+- Support for multiple AI providers (OpenAI, other Azure Cognitive Services)
+- Caching of AI suggestions to reduce API calls
+- Customizable prompt templates
+- Batch processing for multiple risks
+- Integration with Azure Key Vault for credential management
```

Comment:
This may require a review based on the backend-only configuration changes we made.